message_definitions/v1.0/common.xml

```diff
@@ -4714,6 +4714,7 @@
       <field type="float" name="horizontal_fov" units="rad">Horizontal Field of View (angle) where the distance measurement is valid and the field of view is known. Otherwise this is set to 0.</field>
       <field type="float" name="vertical_fov" units="rad">Vertical Field of View (angle) where the distance measurement is valid and the field of view is known. Otherwise this is set to 0.</field>
       <field type="float[4]" name="quaternion">Quaternion of the sensor orientation in vehicle body frame (w, x, y, z order, zero-rotation is 1, 0, 0, 0). Zero-rotation is along the vehicle body x-axis. This field is required if the orientation is set to MAV_SENSOR_ROTATION_CUSTOM. Set it to 0 if invalid."</field>
+      <field type="uint8_t" name="signal_quality" units="%">Signal quality of the sensor. Specific to each sensor type, representing the relation of the signal strength with the target reflectivity, distance, size or aspect, but normalised as a percentage. 0 = unknown/unset signal quality, 1 = invalid signal, 100 = perfect signal.</field>
```

Comment:
why can't we use the covariance?

Reply:
@hamishwillee can you comment on tridge?

Reply:
Nope. Perhaps @auturgy can remember this discussion?

Reply:
@julian.oes Can you remember this? I can dig, but probably you have a better view.

Reply:
I guess the issue here is not the overhead when signal_quality = 0.
The problem is the overhead when signal_quality != 0 and (horizontal_fov = 0 and vertical_fov = 0 and quaternion = {0} )
Is this your point here @tridge ?
@TSC21 is it worth the overhead?

Reply:
I would say that sensor that provides you a signal quality metric that you can use, will usually also provide you hfov and vfov. If it's worth? Depend on how much detailed your sensor data to be.